### DISCUSSION TITLE
What is the difference between the Scanpy API and the Original API?

### DISCUSSION THREAD

**Discussion by suzuki-shm** [brianhie/scanorama#114] [Q&A] [answered]
Hi! Thank you for providing cool tool!

As I asked in the title, I am glad if I could know the difference between [Scanpy API](https://scanpy.readthedocs.io/en/stable/generated/scanpy.external.pp.scanorama_integrate.html) and [Original API](https://github.com/brianhie/scanorama#api-example-usage) .

My understanding is that the Scanpy API requires PCA information before executing a method. The original API, on the other hand, does not seem to require it; the Scanpy documentation states the following:
> This function should be run after performing PCA but before computing the neighbor graph, as illustrated in the example below.

Currently, I plan to use the original API because it does not require the creation of a single anndata object prior to execution. If I created a large AnnData object in advance, I would have to split it up for each batch during preprocessing such as normalization or cell cycle correction. It would be a hassle. So I feel it would be easier to run the preprocessing in separate AnnData objects, put them in a list, and run them in the original API.

If there is anything to keep in mind when using the original API compared to the Scanpy API, I would appreciate it if you could share it with me.

**Comment by brianhie** [comment, marked as answer]
Good question! The only major difference is convenience, you should use the version you're most comfortable with. Both APIs use PCA -- the original API just does PCA under-the-hood, whereas the Scanpy API expects PCA-preprocessed data.

**Comment by suzuki-shm** [comment]
Thanks!
